src/Strategy/Node/Npm/PackageLockV3.hs

```diff
@@ -0,0 +1,435 @@
+--    - http://npm.github.io/how-npm-works-docs/npm3/how-npm3-works.html
+--
+--  --
+buildGraph :: PackageLockV3 -> Graphing Dependency
+buildGraph pkgLockV3 = run . evalGrapher $ do
+  for_ (Map.toList $ packages pkgLockV3) $ \(pkgPathKey, pkgMetadata) -> do
+    case pkgPathKey of
+      -- For root package,
+      --
+      -- "": {
+      --   "name": "project-name",
+      --   "version": "1.0.0",
+      --   "dependencies": {
+      --     "someDirectDep": "^1.0.0",
+      --   }
+      --  },
+      --
+      -- Now this, @someDirectDep, will be resolved at:
+      --    - top path: node_modules/someDirectDep
+      --
+      -- "node_modules/someDirectDep": {
+      --       "version": "1.2.3",
+      --       "resolved": "https://registry.npmjs.org/someDirectDep/-/someDirectDep-1.2.3.tgz",
+      --       "dependencies": {
+      --         "someTransitiveDepB": "^1.0.0"
+      --       }
+      --     },
+      --
+      -- We need to resolve dependency from a top path, since we
+      -- want to identify precise version (e.g. 1.2.3) used by project as opposed
+      -- to version spec (e.g. ^1.0.0) provided in 'dependencies' field.
+      --
+      -- Root's direct dependencies are always resolved at top-path.
+      PackageLockV3Root -> do
+        let directDeps :: [PackagePath]
+            directDeps =
+              toTopLevelPackage
+                <$> concatMap
+                  Map.keys
+                  [ plV3PkgDependencies pkgMetadata
+                  , plV3PkgDevDependencies pkgMetadata
+                  , plV3PkgPeerDependencies pkgMetadata
+                  ]
+
+        let resolvedDeps :: [Dependency]
+            resolvedDeps = mapMaybe toDependency directDeps
+
+        traverse_ direct resolvedDeps
+
+      -- For workspace packages, for instance:
+      --
+      -- "workspace-a": {
+      --   "name": "someWorkspacePkgName",
+      --   "version": "1.0.0",
+      --   "dependencies": {
+      --     "someWorkspaceDirectDep": "^1.0.0",
+      --   }
+      --  },
+      --
+      -- Now this, @someWorkspaceDirectDep, might be resolved at:
+      --    - top path: node_modules/someWorkspaceDirectDep
+      --    - vendor path: workspace-a/node_modules/someWorkspaceDirectDep
+      --
+      -- If the @someWorkspaceDirectDep was resolved with top path,
+      -- vendored path (within workspace-a) will not exist. For this reason,
+      -- always prefer vendored path, if such path does not exist fallback to
+      -- top level path.
+      --
+      -- Direct Dependencies within workspace pkg are treated as direct.
+      -- Direct Peer Dependencies within workspace pkg are treated as direct.
+      -- Development Dependencies within workspace pkg are treated as direct development.
+      PackageLockV3WorkSpace workspaceRootPath -> do
+        let directDeps :: [PackagePath]
+            directDeps =
+              (vendoredPathElseTopLevelPath workspaceRootPath)
+                <$> concatMap
```

Comment:
[required]: Use `map` instead of `<$>` or `fmap` for lists:
```suggestion
              map (vendoredPathElseTopLevelPath workspaceRootPath) $
                concatMap
```